mod-users-bl/src/test/java/org/folio/ModUsersBLTests.java

```diff
@@ -0,0 +1,31 @@
+package org.folio;
+
+import org.folio.test.TestBase;
+import org.folio.test.config.TestModuleConfiguration;
+import org.folio.test.services.TestIntegrationService;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class ModUsersBLTests extends TestBase {
+  private static final String TEST_BASE_PATH = "classpath:domain/mod-users-bl/features/";
+
+  public ModUsersBLTests() {
+    super(new TestIntegrationService(new TestModuleConfiguration(TEST_BASE_PATH)));
+  }
+
+  @BeforeAll
+  public void setup() {
+    runFeature("classpath:domain/mod-users-bl/users-bl-junit.feature");
+  }
+
+  @AfterAll
+  public void tearDown() {
+    runFeature("classpath:common/destroy-data.feature");
+  }
```

Comment:
Does this mean that after all of the tests in this test are run, all of the data set up by the common part of the Karate tests are deleted?

Reply:
There doesn't seem to be a corresponding common function for creating data. What triggers the creation of the tenant etc?

Reply:
I think the call chain is:

1. JUnit invokes the setup method in ModUsersBLTests.java which calls users-bl-junit.feature
2. users-bl-junit.feature calls common/setup-users.feature
3. setup-users.feature calls common/tenant.feature@create etc

One nice thing is that the each module runs in its own space, creating its own tenant, registering the modules it needs, and then cleans up after itself. So there's no danger of collisions between tests.

Reply:
When I say "each module" I mean each sub-project in the main repo. The modules that get registered for the tenant that gets created are defined in users-bl-junit.feature.

Reply:
> When I say "each module" I mean each sub-project in the main repo.

I understand that. This confusion of terminology is caused by the decision to organise these tests by module rather than capability.

Reply:
> 1. JUnit invokes the setup method in ModUsersBLTests.java which calls users-bl-junit.feature
> 2. users-bl-junit.feature calls common/setup-users.feature
> 3. setup-users.feature calls common/tenant.feature@create etc
 
Thank you for this explanation.

What I don't think I understand is the asymmetry. The setup code is defined in the feature, yet the tear down code is defined in the JUnit test.

> So there's no danger of collisions between tests.

I think you mean between tests in other sub-projects. Tests in the same sub-project could presumably clash with each other because their share state.

Reply:
Theres a setup() method in `ModPermsisionsTests.java` that controls the setup too. So it's all flowing out of the java file, setup and teardown. So I guess it's symmetrical in that sense. The organization of the features is another matter though. It's really up to us to determine that for our sub-project, but otherwise I'm taking what has been given (the organization of stuff in `common`).

Reply:
> Theres a setup() method in `ModPermsisionsTests.java` that controls the setup too. So it's all flowing out of the java file, setup and teardown. 

Did you mean `ModUsersBLTests.java`?


> So I guess it's symmetrical in that sense. The organization of the features is another matter though. It's really up to us to determine that for our sub-project, but otherwise I'm taking what has been given (the organization of stuff in `common`).

I understand taking an approach that fits with the information you've been given. I'm playing catch up and trying to understand it. That may well be an unfair burden to put on you for offering to help. 

(Within the FOLIO community, I'm also a bit of a radical when it comes to my beliefs about software development)